lib/shared-method.js

```diff
@@ -369,7 +369,9 @@ function coerceAccepts(uarg, desc) {
   if (actualType === 'number' && Number.isNaN(uarg)) {
     throw new badArgumentError(name + ' must be a number');
   }
-
+  if (actualType === 'integer' && Number.isNaN(uarg)) {
```

Comment:
> No, as there is already a check for the data type at line#352.

Is my understanding correct that for non-integer numeric value, `actualType` will be `number`, and thus we will reject the value on the grounds that `number` is not `integer`? That makes sense.

Please add a unit-test to cover this case.